class.c

```diff
@@ -0,0 +1,1032 @@
+/*    class.c
+ *
+ *    Copyright (C) 2022 by Paul Evans and others
+ *
+ *    You may distribute under the terms of either the GNU General Public
+ *    License or the Artistic License, as specified in the README file.
+ *
+ */
+
+/* This file contains the code that implements perl's new `use feature 'class'`
+ * object model
+ */
+
```

Comment:
This should include:
```
#define PERL_IN_CLASS_C
```
and likely at least some definitions in embed.fnc should be restricted with it. See all our other .c files for similar boilerplate.

Reply:
Ahyes, added.